modules/builds-about.adoc

```diff
@@ -0,0 +1,59 @@
+// Module included in the following assemblies:
+//
+// * assembly/understanding-builds
+
+[id='builds-about-{context}']
+= What is a build?
+
+A _build_ is the process of transforming input parameters into a resulting
+object (build object). Most often, the process is used to transform input
+parameters or source code into a runnable container image.
+
+A _build configuration_, or `BuildConfig`, is characterized by a _build strategy_
+and one or more sources. The strategy determines the aforementioned process, while
+the sources provide its input.
+
+{product-title} leverages Kubernetes by creating containers
+from build images and pushing them to a container image registry.
+
+Builds take advantage of resource restrictions, specifying limitations on
+resources such as CPU usage, memory usage, and build or pod execution time. Build
+objects share common characteristics including inputs for a build, need to
+complete a build process, logging the build process, and publishing resources from
+successful builds and the final status of the build.
+
+The {product-title} build system provides extensible support for _build
+strategies_ that are based on selectable types specified in the build API.
+
+*The build strategies are:*
+
+* Docker build
+* Source-to-Image (S2I) build
+* Custom build
+
+*The six types of sources that can be given as _build input_ are:*
+
+ifndef::openshift-online[]
+* Inline Dockerfile definitions
```

Comment:
If "Dockerfile" is only in online, should Docker be listed as an available build strat for all versions?

Reply:
It's in everything but (was `ifndef` changed to `ifdef`)